src/main/java/org/chrisoft/jline4mcdsrv/Console.java

```diff
@@ -49,33 +60,31 @@ public void append(LogEvent event) {
                         lr.getTerminal().writer().flush();
                     }
                 };
-                conappender.start();
+                conAppender.start();
 
-                org.apache.logging.log4j.core.Logger l = (org.apache.logging.log4j.core.Logger) LogManager.getLogger();
-                l.removeAppender(l.getAppenders().get("SysOut"));
+                //replace SysOut appender with conAppender
                 LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
-                LoggerConfig conf = ctx.getConfiguration().getLoggerConfig(LogManager.getLogger().getName());
-                Level level = conf.getLevel();
-                conf.addAppender(conappender, level, null);
+                LoggerConfig conf = ctx.getConfiguration().getLoggerConfig(logger.getName());
+                conf.removeAppender("JLine"); //only used to set log pattern
+                conf.removeAppender("SysOut");
+                conf.addAppender(conAppender, conf.getLevel(), null);
                 ctx.updateLoggers();
 
-                String s;
-                while (true) {
+                while (!srv.isStopped() && srv.isRunning()) {
                     try {
-                        s = lr.readLine("/");
+                        String s = lr.readLine("/");
                         srv.enqueueCommand(s, srv.getCommandSource());
                         if (s.equals("stop"))
                             break;
-                    }
-                    catch (UserInterruptException e) {}
-                    catch (EndOfFileException e) {
+                    } catch (EndOfFileException|UserInterruptException e) {
                         srv.enqueueCommand("stop", srv.getCommandSource());
-                        return;
+                        break;
                     }
```

Comment:
Catching `UserInterruptException` as well allows to stop the server via Ctrl+C or equivalent